daemon/api_notices.go

```diff
@@ -31,7 +31,7 @@ var (
 	noticesCmd = &Command{
 		Path:       "/v2/notices",
 		GET:        getNotices,
-		ReadAccess: openAccess{},
+		ReadAccess: noticeReadAccess{},
```

Comment:
Does `getNotice` also need to be changed to `ReadAccess: noticeReadAccess{}` rather than the existing `openAccess{}`?

Reply:
I implemented this in separate commit https://github.com/snapcore/snapd/pull/13619/commits/6b16c2d9082eb34d091e094c99c35c59b8a25f12, in case we want to revert it. I opted to do this inside the api handler instead of a separate (or modified) access checker to avoid an extra state lock, but I don't like my approach here that much.

@bboozzoo  @pedronis would like to know your opinions.